common/static/common/js/discussion/utils.js

```diff
@@ -444,19 +463,52 @@
             }
         };
 
-        DiscussionUtil.abbreviateHTML = function(html, minLength) {
+        DiscussionUtil.convertMath = function(element, alreadyEscaped) {
+            var elementContent = element.html();
+            if (alreadyEscaped){
+                // markdownWithHighlight requires unescaped text, so we must unescape it.
+                // safe-lint: disable=javascript-escape
+                elementContent = _.unescape(edx.HtmlUtils.HTML(elementContent));
+            }
+            edx.HtmlUtils.setHtml(
+                element,
+                this.postMathJaxProcessor(this.markdownWithHighlight(elementContent))
+            );
+        };
+
+        DiscussionUtil.abbreviateHTML = function(htmlSnippet, maxLength) {
             var $result, imagesToReplace, truncated_text;
-            truncated_text = jQuery.truncate(html, {
-                length: minLength,
+            truncated_text = edx.HtmlUtils.HTML(jQuery.truncate(htmlSnippet.toString(), {
+                length: maxLength,
                 noBreaks: true,
                 ellipsis: gettext('…')
-            });
-            $result = $("<div>" + truncated_text + "</div>");
+            }));
+            $result = $(edx.HtmlUtils.joinHtml(
+                edx.HtmlUtils.HTML("<div>"),
+                truncated_text,
+                edx.HtmlUtils.HTML("</div>")
+            ).toString());
             imagesToReplace = $result.find("img:not(:first)");
             if (imagesToReplace.length > 0) {
-                $result.append("<p><em>Some images in this post have been omitted</em></p>");
+                edx.HtmlUtils.append(
+                    $result,
+                    edx.HtmlUtils.interpolateHtml(
+                        edx.HtmlUtils.HTML("<p><em>{text}</em></p>"),
+                        {text: gettext("Some images in this post have been omitted")}
+                    )
+                );
             }
-            imagesToReplace.replaceWith("<em>image omitted</em>");
+            // has to do htmlSnippet + ensureHtml because linter still reports javascript-jquery-insertion
+            // in the following (valid?) cases:
+            // 1. imagesToReplace.after(afterMessage.toString())
+            // 2. imagesToReplace.after(edx.HtmlUtils.interpolateHtml(
+            //     edx.HtmlUtils.HTML("<em>{text}</em>"), {text: gettext("image omitted")}
+            // ).toString())
+            // It does not report if it is written on single line, but than it becomes line too long jshint violation
```

Comment:
@robrap I believe the next guy to touch this code will try to optimize it and hit the issue I've faced, so I would rather keep the comment explaining why the code is a bit more complicated than it should be.